maas-region/src/charm.py

```diff
@@ -265,8 +273,25 @@ def _update_ha_proxy(self) -> None:
                             [],
                         )
                     ],
-                }
+                },
             ]
+            if self.config["tls_mode"] == "termination":
+                data.append(
+                    {
+                        "service_name": "agent_service",
+                        "service_host": "0.0.0.0",
+                        "service_port": MAAS_PROXY_PORT,
```

Comment:
since this is adding an item to `data` I suppose that HA proxy is also exposing plain HTTP endpoints? These 2 items are using the same port `MAAS_PROXY_PORT`, is this correct?

Reply:
This is the per maas-region unit relation data. Those data are placed at the unit databags. In addition to that, the user has to provide the services yaml at haproxy charm. In the end, all those data will be merged, giving priority to haproxy configuration option: https://charmhub.io/haproxy/configurations#services

The default service of haproxy has name `haproxy_service` as you can see which is using port 80. So if we relate maas-region to haproxy without TLS (option `""`), maas-region charm only adds the servers to the haproxy backend, keeping the service in plain HTTP.

With TLS "termination", we use the default service `haproxy_service` for user interaction and the `agent_service` for communication between racks/regions. The first one will get TLS from haproxy `services` config, the second one will remain plain HTTP but with an acl. This is how maas-anvil is adding TLS on haproxy charm: https://github.com/canonical/maas-anvil/blob/main/anvil-python/anvil/commands/haproxy.py#L216-L235

- `haproxy_service`: port 443, default certificate (we are adding it through haproxy [config options](https://charmhub.io/haproxy/configurations#ssl_cert)), and http requests redirected to https
- `agent_service`: port 80 still but with an ACL to allow incoming traffic only from the management network.